sunpy/net/jsoc/jsoc.py

```diff
@@ -489,165 +497,131 @@ def _process_time(self, time):
-
-        This routine puts in a POST request to JSOC
-        """
+            elif pkey == 'WAVELNTH' and wavelength is not '':
+                if isinstance(wavelength, list):
+                    wavelength = [int(np.ceil(wave.to(u.AA).value)) for wave in wavelength]
+                    wavelength = str(wavelength)
+                else:
+                    wavelength = '[{0}]'.format(int(np.ceil(wavelength.to(u.AA).value)))
+                pkstr += wavelength
 
-        payload = self._make_query_payload(start_time, end_time, series,
-                                           notify=notify, protocol=protocol,
-                                           compression=compression, **kwargs)
+            elif len(primekeys) > 0:
+                pkstr += '[{0}]'.format(primekeys.pop(pkey, ''))
 
-        r = requests.post(JSOC_EXPORT_URL, data=payload)
+            else:
+                break
 
-        if r.status_code != 200:
-            exception_message = "JSOC POST Request returned code {0}"
-            raise Exception(exception_message.format(r.status_code))
+        dataset = '{series}{primekeys}{segment}'.format(series=series,
+                                                        primekeys=pkstr,
+                                                        segment=segment)
 
-        return r, r.json()
+        return dataset
 
     def _lookup_records(self, iargs):
         """
         Do a LookData request to JSOC to workout what results the query returns
         """
-        keywords = ['DATE', 'TELESCOP', 'INSTRUME', 'T_OBS', 'WAVELNTH',
-                    'WAVEUNIT']
+
+        keywords_default = ['DATE', 'TELESCOP', 'INSTRUME', 'T_OBS', 'WAVELNTH']
+        isMeta = iargs.get('meta', False)
+        c = drms.Client()
+
+        if isMeta:
+            keywords = '***ALL***'
+        else:
+            keywords = iargs.get('keys', keywords_default)
 
         if not all([k in iargs for k in ('start_time', 'end_time', 'series')]):
             error_message = "Both Time and Series must be specified for a "\
                             "JSOC Query"
             raise ValueError(error_message)
 
+        if not isinstance(keywords, list) and not isinstance(keywords, six.string_types):
+            error_message = "Keywords can only be passed as a list or"\
+                            " comma-separated strings."
+            raise ValueError(error_message)
+
+        segments = iargs.get('segment', '')
+        if segments:
+            if not isinstance(segments, list) and not isinstance(segments, six.string_types):
+                error_message = "Segments can only be passed as a list or"\
+                                " comma-separated strings."
+                raise ValueError(error_message)
+
+        pkeys = c.pkeys(iargs['series'])
+        pkeys_passed = iargs.get('primekeys', None)
+        if pkeys_passed is not None:
+            if not set(list(pkeys_passed.keys())) < set(pkeys):
```

Comment:
Yes, `pkeys_passed` is a dict. I will add the comment.
`set(a) < set(b)` returns `True` if `a` is a subset of `b`. Hence, if a and b are same, it will return `True`.